morebits.js

```diff
@@ -3696,37 +3694,59 @@ Morebits.wiki.page = function(pageName, currentAction) {
 
 		var pageTitle = $(xml).find('page').attr('title');
 
-		// fetch existing protection levels
+		// Fetch existing protection levels
 		var prs = $(xml).find('pr');
-		var editprot = prs.filter('[type="edit"]');
+		var editprot = prs.filter('[type="edit"]:not([source])');
 		var moveprot = prs.filter('[type="move"]');
 		var createprot = prs.filter('[type="create"]');
 
-		var protections = [], expirys = [];
+		// Fall back to current levels if not explicitly set
+		if (!ctx.protectEdit && editprot.length) {
+			ctx.protectEdit = { level: editprot.attr('level'), expiry: editprot.attr('expiry') };
+		}
+		if (!ctx.protectMove && moveprot.length) {
+			ctx.protectMove = { level: moveprot.attr('level'), expiry: moveprot.attr('expiry') };
+		}
+		if (!ctx.protectCreate && createprot.length) {
+			ctx.protectCreate = { level: createprot.attr('level'), expiry: createprot.attr('expiry') };
+		}
+
+		// Warn if cascading protection being applied with an invalid protection level,
+		// which for edit protection will cause cascading to be silently stripped
+		// Also default to pre-existing cascading protection if unchanged (as with others)
+		if (ctx.protectCascade || (ctx.protectCascade === null && !!prs.filter('[cascade]').length)) {
+			// On move protection, this is technically stricter than the MW API,
+			// but seems reasonable to avoid dumb values and misleading log entries (T265626)
+			if (((!ctx.protectEdit || ctx.protectEdit.level !== 'sysop') ||
+				(!ctx.protectMove || ctx.protectMove.level !== 'sysop')) &&
+				!confirm('You have cascading protection enabled on "' + ctx.pageName +
```

Comment:
We could also just remove the `confirm` and error out; there's kind of an XY situation as to whether converting to sysop or removing cascading is the intent, although presumably the former is indeed more likely.